pyvo/dal/tap.py

```diff
@@ -77,16 +77,25 @@ class TAPService(DALService, AvailabilityMixin, CapabilityMixin):
 
     _tables = None
 
-    def __init__(self, baseurl):
+    def __init__(self, baseurl, session=None):
         """
         instantiate a Tablee Access Protocol service
 
         Parameters
         ----------
         baseurl :  str
            the base URL that should be used for forming queries to the service.
+        session : object
+           optional session to use for network requests
         """
-        super().__init__(baseurl)
+        super().__init__(baseurl, session=session)
+
+        # Check if the session has a parse_capabilities attribute.
+        # This means that the session is aware of IVOA capabilities,
+        # and can use this information in processing network requests.
+        # One such usecase for this is auth.
+        if hasattr(self._session, 'parse_capabilities'):
+            self._session.parse_capabilities(self.capabilities)
```

Comment:
Not sure about this one, there is already a method named `parse_capabilities` in the xml code.

Reply:
Well nobody has ever said I'm overly creative about variable names. :smile:
I've gone ahead and changed this to update_from_capabilities

Reply:
Was just something which confused me at some point.